From a8baf318bb7d5fa474e30cb3a28ba3d0b2159c00 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:07 +0800
Subject: [PATCH 090/150] ACPI: APD: Add clock frequency for Phytium Pe220x I2C
 controller

Add the input clock frequency of 50MHz for Phytium I2C for Pe220x SoCs.
The ACPI ID is "PHYT0038".

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I5ef61779e27e86fd69277fc404b740c3098119c4
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 drivers/acpi/acpi_apd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/acpi/acpi_apd.c b/drivers/acpi/acpi_apd.c
index e0fe87feb175..9e265b3ba606 100644
--- a/drivers/acpi/acpi_apd.c
+++ b/drivers/acpi/acpi_apd.c
@@ -167,6 +167,11 @@ static const struct apd_device_desc phytium_i2c_desc = {
 	.fixed_clk_rate = 200000000,
 };
 
+static const struct apd_device_desc phytium_pe220x_i2c_desc = {
+	.setup = acpi_apd_setup,
+	.fixed_clk_rate = 50000000,
+};
+
 static const struct apd_device_desc thunderx2_i2c_desc = {
 	.setup = acpi_apd_setup,
 	.fixed_clk_rate = 125000000,
@@ -252,6 +257,7 @@ static const struct acpi_device_id acpi_apd_device_ids[] = {
 	{ "HISI0173", APD_ADDR(hip08_spi_desc) },
 	{ "NXP0001", APD_ADDR(nxp_i2c_desc) },
 	{ "PHYT0003", APD_ADDR(phytium_i2c_desc) },
+	{ "PHYT0038", APD_ADDR(phytium_pe220x_i2c_desc) },
 #endif
 	{ }
 };
-- 
2.47.0

